zenodo_uploader.py

```diff
@@ -1,81 +1,278 @@
 import os
-
+import hashlib
 import argparse
 import requests
 import yaml
 import json
+from urllib.parse import urlencode
+import traceback
 
-def upload_to_zenodo(
-    file_name_to_upload,
-    config_file,
-    zenodo_server="https://sandbox.zenodo.org/api/deposit/depositions",
+def search_for_deposition(
+    title,
+    owner=None,
+    zenodo_server="https://sandbox.zenodo.org/api/",
 ):
-    filename = os.path.abspath(file_name_to_upload)
-    if not os.path.isfile(filename):
-        raise FileNotFoundError(
-            f"The file, specified for uploading does not exist: {filename}"
-        )
-        
-    config_name = os.path.abspath(config_file)
-    if not os.path.isfile(config_name):
-        raise FileNotFoundError(
-            f"The file with metadata, specified for uploading does not exist: {config_name}"
-        )
-        
-    headers = {"Content-Type": "application/json"}
-    params = {"access_token": os.getenv("ZENODO_ACCESS_TOKEN", "")}
+    print(f"Searching for depositions...\n")
+    search = f'metadata.title:"{title}"'
+    if owner:
+        search += f" owners:{owner}"
+
+    search = search.replace("/", " ")  # zenodo can't handle '/' in search query
+
+    params = {"q": search, "sort": "mostrecent"}
+    url = zenodo_server + "deposit/depositions?" + urlencode(params)
```

Comment:
We use the style of f-strings here, so I suggest converting it to that:
```suggestion
    url = f"{zenodo_server}deposit/depositions?{urlencode(params)}"
```